docs/dev-manuals/tooling/techstack-and-service-creation.md

```diff
@@ -0,0 +1,64 @@
+---
+sidebar_position: 3
+---
+
+import CustomImage from "@site/src/components/CustomImage";
+
+# Techstack and service creation
+
+### Techstack pverview
+
+This is a general overview of some of the technologies used in MiSArch and which services / components they involve.
+
+<CustomImage path="/images/techstack-overview" width="2803" height="2341" />
+
+A user client sends a request to Nginx to retrieve the frontend. The frontend then communicates with the backend through Nginx. Nginx acts both as a web-server and reverse proxy. The frontend creates requests in GraphQL. The GraphQL gateway resolves the request by querying the respective services.
+
+Dapr acts as a middleware abstracting communication between services. This enables metric and trace collection via Open Telemetry. Services communicate via Dapr sidecars. Each sidecar is associated with one service and known by the Dapr placement / master service, also known as Dapr Operator Service. The Dapr Operator feeds the Open Telemetry collector with metrics and traces.
+
+A full Zipkin stack stores and displays traces, while metrics are collected by Grafana / Prometheus. Each Service involved in GraphQL queries needs to expose a GraphQL compatible API. Each services needs to be registered in the GraphQL gateway.
+
+### Infastructure (as Code)
+
+The current plan is to support two deployments:
+
+1. Kubernetes
+
+   - State-of-the-art container orchestration. Widely used for microservices in science and the industry. Powerful but tedious to set up. Supports multi-node deployments, key stores, load balancing, auto scaling, etc.
+
+2. Docker Compose
+   - Single-node deployments. Good portability + easy setup. Declarative multi-container app tool. Packaged with Docker
+
+**Why two deployments?**
+
+Docker Compose assists development by enabling faster code + deployment iterations than Kubernetes. Kubernetes is a requirement and therefore MiSArch needs to be deployed with Kubernetes from time-to-time. We do not want to big-bang port MiSArch to Kubernetes at the end of the project. Incremental ports help to detect deployment difficulties early on.
+
+**(as Code)?**
+
+We plan on using Terraform for Kubernetes to deploy in a declarative way, which enables to version our deployments according to the infrastructure as code paradigm. Docker Compose might also benefit from Terraform as it enables configuration manipulation while abstracting boiler-plate code. This can for example reduce port mapping conflicts.
```

Comment:
did you do some reasearch if docker compose is reasonably supported with terraform? I'm not sure

Reply:
It looks like Terraform supports Docker, but not Docker Compose https://developer.hashicorp.com/terraform/tutorials/docker-get-started/docker-build. Makes sense as Docker Compose is already declarative on its own. The question is if we gain advantages by using Terraform + Docker in comparison to Docker Compose.

Also, if Terraform makes it less tedious by being declarative, why not only support Terraform + Kubernetes and drop Docker Compose?

Reply:
we decided in call to support both, as docker compose is helpful for developing and also easier to deploy for reasearchers when trying the reference architecture